api/v1/ytsaurus_types.go

```diff
@@ -92,6 +92,14 @@ type LocationSpec struct {
 	// Disk space quota, default is size of related volume.
 	//+optional
 	Quota *resource.Quantity `json:"quota,omitempty"`
+	// Location watermark threshold in percents of volume size.
+	//+kubebuilder:default:=10
+	//+kubebuilder:validation:Minimum:=1
+	//+kubebuilder:validation:Maximum:=50
+	Watermark int32 `json:"watermark,omitempty"`
```

Comment:
Do I understand correctly that 10% was hardcoded and now it is the default in spec?

Reply:
Yes, you are absolutely right.

Reply:
Maybe a better approach would be to have a pointer here and in case of nil put this 10% default in _code_, that way we don't bother users with diff after updating the operator and we are free to change the default value in the future.

Reply:
You mean to do smth like that:
```go
if location.Watermark != nil {
   watermark = ptr.Int32(location.Watermark.Value());
} else {
   watermark = 10;
}
```
In fact, I intentionally made explicit default to make its value transparent for the users. And we should bother them with diff in case of changing the default as it affects the watermarks. I definitely want to be bothered by such change 😄 .
For me, frankly saying, it's much better to see explicit defaults rather than to search them in the source code of YTsaurus for all optional fields. But at least here we have to keep the default since this parameter is operator specific only.

Reply:
I think, that in the ideal world, default should be only mentioned in the docs (+release notes if changed) and the user shouldn't have it in his spec and see the diff and so on. Although maybe you're right, and some users don't want to read release notes (or we can forget to mention some stuff there) and prefer to see diff in the spec. I'll discuss that with other maintainers.

In this particular case it is kinda confusing for user to see the diff if in reality the value was not changed.

This is minor thing anyway since we don't have some policy here, you can left it as it is, we explain diff in the release notes.